crypto/x509/x509_cmp.c

```diff
@@ -391,15 +391,12 @@ int X509_check_private_key(const X509 *x, const EVP_PKEY *k)
     int ret;
 
     xk = X509_get0_pubkey(x);
+    if (xk == NULL) {
+        ERR_raise(ERR_LIB_X509, X509_R_UNABLE_TO_GET_CERTS_PUBLIC_KEY);
```

Comment:
Please note that in X509_PUBKEY_get0 there is already `ERR_raise(ERR_LIB_EVP, EVP_R_DECODE_ERROR);` Which is of course kind of awkward message so I am not saying this one should be dropped. Perhaps the EVP_R_DECODE_ERROR in X509_PUBKEY_get0 should be simply replaced with this one?

Reply:
Good suggestion.  There is some additional clean up possible in these functions.